R/latent_individual.R

```diff
@@ -108,28 +108,81 @@ is_latent_individual <- function(data) {
   inherits(data, "epidist_latent_individual")
 }
 
+#' Check if data has the `epidist_latent_individual` class
+#'
+#' @param data A `data.frame` or `data.table` containing line list data
+#' @param family Output of a call to `brms::brmsfamily()`
+#'
+#' @importFrom rstan lookup
+#' @method epidist_family epidist_latent_individual
+#' @family latent_individual
+#' @export
+epidist_family.epidist_latent_individual <- function(data,
+                                                     family = brms::lognormal(),
+                                                     ...) {
+  epidist_validate(data)
+  if (!checkmate::test_class(family, "family")) {
+    cli::cli_abort("family must be a family object")
+  }
+  family <- brms:::validate_family(family) # allows use of stats:: family
+  # brms also has lb and ub in brms::custom_family, but not in brms::brmsfamily
+  non_mu_links <- family[[paste0("link_", setdiff(family$dpars, "mu"))]]
+  # Use the function dpar_bounds() here
+  brms::custom_family(
+    paste0("latent_", family$family),
+    dpars = family$dpars,
+    links =   c(family$link, non_mu_links),
+    type = family$type,
+    vars = c("pwindow", "swindow", "vreal1"),
+    loop = FALSE
+  )
+}
+
 #' Define a formula for the latent_individual model
 #'
 #' @param data ...
-#' @param delay_central Formula for the delay mean. Defaults to intercept only.
-#' @param sigma Formula for the delay standard deviation. Defaults to intercept
-#' only.
+#' @param family The output of `epidist_family()`
```

Comment:
Should this instead be a `brmsfamily` object? Could still extract out the `dpars`